### PR TITLE
Improve query profiler

### PR DESCRIPTION
**Pull request by vbisserie**
- Bugfix on tables path verification
- Search under <config file path>.d directory if exists in order to parse all sub configuration files
- In pack definition, check if it is an internal one and parse it

### CODE REVIEW

File: tools/tests/utils.py [hunk 1]
```diff
@@ -95,18 +95,18 @@ def queries_from_config(config_path):
...
     if "packs" in config:
         for keys,values in config["packs"].iteritems():
-            with open(values) as fp:
-                packfile = fp.read()
-                packcontent = rmcomment.sub('',packfile)
-                packqueries = json.loads(packcontent)
-                for queryname,query in packqueries["queries"].iteritems():
+            # Check if it is an internal pack definition
+            if type(values) is dict:
+                for queryname,query in values["queries"].iteritems():
```
Comment (theopolis) [inline tools/tests/utils.py]: nit, space between `,`

File: tools/tests/utils.py [hunk 2]
```diff
@@ -95,18 +95,18 @@ def queries_from_config(config_path):
...
                     queries["pack_"+queryname] = query["query"]
+            else:
+                with open(values) as fp:
+                    packfile = fp.read()
+                    packcontent = rmcomment.sub('',packfile)
```
Comment (theopolis) [inline tools/tests/utils.py]: nit, space

File: tools/tests/utils.py [hunk 3]
```diff
@@ -95,18 +95,18 @@ def queries_from_config(config_path):
...
+                    packqueries = json.loads(packcontent)
+                    for queryname,query in packqueries["queries"].iteritems():
+                        queries["pack_"+queryname] = query["query"]
```
Comment (theopolis) [inline tools/tests/utils.py]: nit, space between `+` and operands

File: tools/tests/utils.py [hunk 4]
```diff
@@ -95,18 +95,18 @@ def queries_from_config(config_path):
...
+            # Check if it is an internal pack definition
+            if type(values) is dict:
+                for queryname,query in values["queries"].iteritems():
                     queries["pack_"+queryname] = query["query"]
+            else:
+                with open(values) as fp:
+                    packfile = fp.read()
+                    packcontent = rmcomment.sub('',packfile)
+                    packqueries = json.loads(packcontent)
+                    for queryname,query in packqueries["queries"].iteritems():
```
Comment (theopolis) [inline tools/tests/utils.py]: nit, space after `,`

File: tools/analysis/profile.py [hunk 5]
```diff
@@ -330,7 +330,7 @@ def regress_check(profile1, profile2):
     if not args.force and not os.path.exists(args.shell):
         print("Cannot find --shell: %s" % (args.shell))
         exit(1)
-    if args.config is None and not os.path.exists(args.tables):
+    if args.tables is None and not os.path.exists(args.tables):
```
Comment (theopolis) [inline tools/analysis/profile.py]: Why this change?

Comment (vbisserie) [inline tools/analysis/profile.py]: Sorry, I misunderstood the check

**Comment by theopolis** [comment]
ok to test
